AK/Time.h

```diff
@@ -84,9 +84,35 @@ class Time {
 public:
     Time() = default;
     Time(const Time&) = default;
+    Time& operator=(const Time&) = default;
 
-    static Time from_seconds(i64 seconds) { return Time(seconds, 0); }
-    static Time from_nanoseconds(i32 nanoseconds);
+    Time(Time&& other)
+        : m_seconds(exchange(other.m_seconds, 0))
+        , m_nanoseconds(exchange(other.m_nanoseconds, 0))
+    {
+    }
+    Time& operator=(Time&& other)
+    {
+        if (this != &other) {
+            m_seconds = exchange(other.m_seconds, 0);
+            m_nanoseconds = exchange(other.m_nanoseconds, 0);
+        }
+        return *this;
+    }
+
+    constexpr static Time from_seconds(i64 seconds) { return Time(seconds, 0); }
+    constexpr static Time from_nanoseconds(i64 nanoseconds)
+    {
+        return Time(nanoseconds / 1'000'000'000, nanoseconds % 1'000'000'000);
+    }
+    constexpr static Time from_microseconds(i64 microseconds)
+    {
+        return Time(microseconds / 1'000'000, (microseconds % 1'000'000) * 1'000);
+    }
+    constexpr static Time from_milliseconds(i64 milliseconds)
```

Comment:
Perhaps a silly suggestion, but how about a uom-like interface for these?
`1ms`? it'd probably just call into `Time`.

Reply:
Go for it ;)

Reply:
_/me adds it to the TODO list_